homeassistant/components/sunsynkweb/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for Sunsynk Inverter Web integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow as BaseConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .api import get_bearer_token
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class AuthChecker:
+    """Check authentication in config flow."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize."""
+        self.session = async_get_clientsession(hass)
+
+    async def authenticate(self, username: str, password: str) -> bool:
+        """Test if we can authenticate with the host."""
+        try:
+            await get_bearer_token(self.session, username, password)
+        except KeyError:
+            return False
+        return True
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
is there a reason why you don't have this inline? It seems to be only used once

Reply:
I just kept the structure as templatised, will simplify

Reply:
updated